### PR TITLE
initial case for vm-import-controller

### PR DESCRIPTION
**Pull request by ibrokethecloud**
the PR introduces the initial test cases for vm-import-controller for rc1

### CODE REVIEW

File: docs/content/manual/_incoming/2274-vm-import.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+export OS_USERNAME="openstack-username"
+export OS_PASSWORD="openstack-password"
+export OS_VM_NAME="openstack-export-test-vm-name"
+export OS_REGION_NAME="openstack-region"
+export KUBECONFIG="kubeconfig-for-harvester-cluster"
+```
+
+## Expected Results
+- VirtualMachines should be imported from Vmware / Openstack to Harvester cluster.
+- Source VirtualMachines should be powered off
```
Comment (khushboo-rancher) [inline docs/content/manual/_incoming/2274-vm-import.md:243]: VM config and data should be copied as it is.

**Review by khushboo-rancher — COMMENTED** [review]
Thanks @ibrokethecloud for detailed steps.
@irishgordo Can you please help review this and help adding more test cases like importing multiple vms, importing vms with combination of network interfaces, CPU, setting etc, testing this with 3 nodes harvester. Interrupting the migration first then completing it, doing some post operations on the migrated VMs etc.

**Comment by irishgordo** [comment]
@khushboo-rancher -I'll dig into this and review this soon :smile:

File: docs/content/manual/_incoming/2274-vm-import.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+* Harvester version: v1.1.0-rc1
+* Vsphere: 7.0
+* Openstack: Simulated using running devstack
+* Download kubeconfig for harvester cluster
+
+## Environment Setup
+
+1. Prepare Harvester master node
+2. Prepare vsphere setup (or use existing setup)
+3. Prepare a devstack cluster (Openstack 16.2)
```
Comment (irishgordo) [inline docs/content/manual/_incoming/2274-vm-import.md:22]: @ibrokethecloud is this the `stable/yoga` branch?
As I was trying to provision a `stable/yoga` instance via some #cloud-config [here](https://gist.github.com/irishgordo/b12f35e02b8e4c52620f546b189f5757)

**Comment by irishgordo** [comment]
@khushboo-rancher - I'm adding some additional cases to this still.
Did merge in some recently.
I'll be opening another PR against the feature branch.

**Comment by irishgordo** [comment]
@khushboo-rancher with multiple network interfaces I've noticed this an opened an issue but will refine the cases to mention nic interfaces:
https://github.com/harvester/harvester/issues/2890#issue-1400437398

File: docs/content/manual/_incoming/2274-vm-import.md (hunk 2 above)
Comment (irishgordo) [reply docs/content/manual/_incoming/2274-vm-import.md]: Doesn't need to be stable yoga per say, as even most recent version of openstack works well "zed"

**Review by khushboo-rancher — APPROVED** [review]
LGTM
Thanks @irishgordo @ibrokethecloud

**Pull request merged**